docs/pages/en-us/Error-Handling.md

```diff
@@ -0,0 +1,114 @@
+---
+layout: default
+title: Error Handling
+
+banner:
+    icon: 'assets/img/hard-hat.png'
+    title: Worker Framework
+    subtitle: Cross Platform, Cloud Ready, Microservice Framework
+    links:
+        - title: GitHub
+          url: https://github.com/WorkerFramework/worker-framework
+          
+---
+
+# Error Handling
+
+The purpose of this page is to provide details for the various types of errors and failures that can arise during message processing. A set of rules that every Worker implementation should adhere too is included as well.
+
+## Worker Success
+We start though by describing the expected flow when messages are handled by Worker implementations without any issues at all.
+
+![Success Worker Flow](images/WorkerSuccessFlow.png)
+*Figure 1 - Message successfully processed by Worker*
+
+1. The Worker will receive a message from the messaging input queue, `worker-input-queue` and will start to process it.
+2. The message is successfully processed by the Worker and acknowledged.
+3. A RESULT_SUCCESS response is placed on the messaging output queue, `worker-output-queue` by the worker framework.
+
+Some messages however cannot be handled by the Worker and these are described next.
+
+## Poisoned Messages
+A poisoned message is a message that a worker is unable to handle. The message is deemed poisonous during processing when repeated catastrophic failure of the worker occurs. Regardless of how many times the message is retried, the worker will not be able to handle the message in a graceful manor. The flow of a poisoned message is described in Figure 2 below:
+
+![Poisoned Messages](images/PoisonedMessage.png)
+*Figure 2 - Poisoned message flow*
+
+1. The Worker will receive a message from the messaging input queue, `worker-input-queue` and will start to process it.
+2. The processing of the message causes the worker to die unexpectedly.
+3. The Autoscaler service scales the worker back up again to facilitate a retry.
+4. The original message is re-delivered for retry where the `x-caf-worker-retry` message header is stamped on the message and incremented. Messages which cause a worker to crash are retried up to 10 times by default before the worker gives up trying to process them and marks them as poisoned. Note that the number of permitted retries is configurable within the `RabbitWorkerQueueConfiguration` file for a worker.
+5. Once the permitted number of retries is exceeded, the original message will be acknowledged.
+6. A `RESULT_EXCEPTION` response including the poisoned message details is placed on the messaging output queue `worker-output-queue` by the worker framework when the permitted number of retries is exceeded.
```

Comment:
I think (5) and (6) are in the wrong order - the message shouldn't be acknowledged until the response has been published.

Reply:
Yes agreed